api/src/main/java/org/apache/iceberg/UpdateStatistics.java

```diff
@@ -27,9 +27,19 @@ public interface UpdateStatistics extends PendingUpdate<List<StatisticsFile>> {
    * the snapshot if any exists.
    *
    * @return this for method chaining
+   * @deprecated since 1.8.0, will be removed 1.9.0 or 2.0.0, use setStatistics(statisticsFile).
    */
+  @Deprecated
   UpdateStatistics setStatistics(long snapshotId, StatisticsFile statisticsFile);
 
+  /**
+   * Set the table's statistics file for given snapshot, replacing the previous statistics file for
+   * the snapshot if any exists.
+   *
+   * @return this for method chaining
+   */
+  UpdateStatistics setStatistics(StatisticsFile statisticsFile);
```

Comment:
This introduces a breaking API change, so we typically add a default impl that throws:

```suggestion
  default UpdateStatistics setStatistics(StatisticsFile statisticsFile) {
    throw new UnsupportedOperationException("Setting statistics is not supported");
  }
```

Then you can remove the changes in `revapi.yml`